src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -0,0 +1,266 @@
+
+        # Emergency visits
+        mask_acute_is = (
+            df[models.ISCHEMIC_STROKE_MODEL_NAME] == models.ACUTE_ISCHEMIC_STROKE_STATE_NAME
+        )
+        mask_acute_mi = (
+            df[models.MYOCARDIAL_INFARCTION_MODEL_NAME]
+            == models.ACUTE_MYOCARDIAL_INFARCTION_STATE_NAME
+        )
+        mask_emergency = mask_acute_is | mask_acute_mi
+        visit_emergency = df[mask_emergency].index
+        visitors[data_values.VISIT_TYPES.EMERGENCY] = visit_emergency
+
+        # Scheduled visits
+        mask_scheduled_non_emergency = (
+            (df[self.scheduled_visit_date_column] > (event.time - event.step_size))
+            & (df[self.scheduled_visit_date_column] <= event.time)
+            & (~mask_emergency)
+        )
+        scheduled_non_emergency = df[mask_scheduled_non_emergency].index
+        # Missed scheduled (non-emergency) visits (these do not get re-scheduled)
+        missed_visit = self.randomness.filter_for_probability(
+            scheduled_non_emergency,
+            df.loc[scheduled_non_emergency, self.miss_scheduled_visit_probability_column],
+        )  # pd.Index
+        visitors[data_values.VISIT_TYPES.MISSED] = missed_visit
+        df.loc[missed_visit, self.scheduled_visit_date_column] = pd.NaT  # no re-schedule
+        visit_scheduled = scheduled_non_emergency.difference(missed_visit)
+        visitors[data_values.VISIT_TYPES.SCHEDULED] = visit_scheduled
+
+        # Background visits (for those who did not go for another reason (even missed screenings))
+        maybe_background = df.index.difference(visit_emergency.union(visit_scheduled))
+        utilization_rate = self.background_utilization_rate(maybe_background)
+        visit_background = self.randomness.filter_for_rate(
+            maybe_background, utilization_rate
+        )  # pd.Index
+        visitors[data_values.VISIT_TYPES.BACKGROUND] = visit_background
+
+        df = self.visit_doctor(df, visitors=visitors, event_time=event.time)
+
+        self.population_view.update(
+            df[[self.visit_type_column, self.scheduled_visit_date_column]]
+        )
+
+    def visit_doctor(
+        self,
+        df: pd.DataFrame,
+        visitors: Dict,
+        event_time: pd.Timestamp,
+        min_followup: int = FOLLOWUP_MIN,
+        max_followup: int = FOLLOWUP_MAX,
+    ) -> pd.DataFrame:
+        """Updates treatment plans and schedules followups.
+
+        Arguments:
+            df: state table to update
+            visitors: dictionary of all simulants visiting the doctor and their visit types
+            event_time: timestamp at end of the time step
+            min_followup: minimum number of days out to schedule followup
+            max_followup: maximum number of days out to schedule followup
+        """
+        # Update visit type
+        df[self.visit_type_column] = "none"
+        for k in visitors:
+            df.loc[visitors[k], self.visit_type_column] = df[self.visit_type_column] + f";{k}"
```

Comment:
this is interesting syntax / just curious; the column `self.visit_type_column` is having `;visitor_key` appended to each row, where `visitor_key` is `data_values.VISIT_TYPES.SCHEDULED`?

Reply:
Yep! I don't love it and there's probably cleverer ways to do this, but in the end you get simulants with visit types like:
- "none"
- "none;scheduled"
- "none;missed"
- "none;emergency"
- "none;background"
- "none;missed;background"

It's that last one that kind of pushed me to doing it this way since we need to count missed AND background seperately.